### PR TITLE
NREL - Solar Anywhere data examples with holoviews

### PR DESCRIPTION
**Pull request by PeterDSteinberg**
This PR is a notebook demo using solar radiation time series from the [NREL - NSRDB Solar Anywhere data set](http://rredc.nrel.gov/solar/old_data/nsrdb/).  The dataset downloaded currently is limited to the Wyoming locations and 20 years of hourly data, including clear sky solar radiation and measured radiation and their direct normal, diffuse horizontal, and global horizontal components.

Work in progress.  TODO:

 * [ ] More explanation of `hv.Curve` in notebook.  I have accidentally done `hv.Curve(x,y)` several times when I intended `hv.Curve((x,y))`, and the error string when I get it wrong is a bit wordy.  I'll check docs as well for places to clarify this.
 * [x] More viz options for median +/- high/low quartiles of time averaged averaged radiation
   * [x] Find the best fix for the leap year Julian hour averaging issue (i.e. does the standard average Julian hour or day average get labeled or adjusted for leap year effects: 8760 or 8784 rows in the results dataframe?)
 * [x] Provide some hour of day summary plots not just Julian hour of year as it is now
 * [ ] Provide a spatial summary plot (currently everything is a time series plot at a point in space)
 * [ ] EDIT: Also use this notebook to explain tab completion in notebooks `%%opts` magic command 

Other ideas welcome.  I hope to use this dataset a bit later for [machine learning with `elm`](https://github.com/ContinuumIO/elm), such as ML to infill data gaps in spatial time series.  I'll try to finalize this PR this week, then revisit the topic later.
 
cc @jbednar @gbrener

### CODE REVIEW

**Comment by philippjfr** [comment]
> I have accidentally done hv.Curve(x,y) several times when I intended hv.Curve((x,y)), and the error string when I get it wrong is a bit wordy. I'll check docs as well for places to clarify this.

That's good feedback and we're hoping to improve this asap, see: https://github.com/ioam/holoviews/issues/1127

**Comment by PeterDSteinberg** [comment]
Some screenshots of the viz done here for NREL solar data.  More can be done to present this data (see unfinished checkboxes above).  @jbednar If this notebook looks reasonable to you, I think we can merge now and improve in separate PR(s).  Made #347 as a reminder for me to follow up on this in a month or so.  
screen shot 2017-05-08 at 10 57 17 pm
screen shot 2017-05-08 at 10 57 24 pm
screen shot 2017-05-08 at 11 00 42 pm

**Comment by philippjfr** [comment]
The ugly tab titles should look a lot better once https://github.com/ioam/holoviews/issues/1425 has been addressed.
